File: app/src/main/java/com/example/myschoolwearclient/ui/homeFragment/HomeFragmentViewModel.java
package com.example.myschoolwearclient.ui.homeFragment;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.myschoolwearclient.firebaseClasses.CategoryItem;
import com.example.myschoolwearclient.firebaseClasses.PopularProductItem;
import com.example.myschoolwearclient.firebaseClasses.ProductItem;
import com.example.myschoolwearclient.repo.Repository;

import java.util.ArrayList;
import java.util.List;

public class HomeFragmentViewModel extends ViewModel {

    Repository repository = Repository.getInstance();

    LiveData<ArrayList<CategoryItem>> allCategories = repository.getCategories();

    private LiveData<List<PopularProductItem>> allPopularProducts = repository.getALlPopularProducts();

    LiveData<ArrayList<ProductItem>> allPopularProductItems = repository.getPopularProductItems();
}

File: app/src/main/java/co/in/plantonic/ui/homeFragment/HomeFragmentViewModel.kt
package co.`in`.plantonic.ui.homeFragment

import androidx.lifecycle.ViewModel
import androidx.lifecycle.LiveData
import co.`in`.plantonic.firebaseClasses.ProductItem
import co.`in`.plantonic.repo.LoginRepository
import co.`in`.plantonic.repo.Repository
import co.`in`.plantonic.retrofit.models.home.HomePageBannerResponseModel

class HomeFragmentViewModel : ViewModel() {
    private val repository = Repository.instance
    private val loginRepository = LoginRepository()


    var allCategories = repository?.categories

//    private val allPopularProducts: LiveData<List<PopularProductItem>> = repository.allPopularProducts()

    val allPopularProductItems: LiveData<ArrayList<ProductItem>>? = repository?.popularProductItems()


    fun checkIfUserExists(userId: String?): LiveData<Boolean?> {
        loginRepository.checkIfUserExists(userId!!)
        return loginRepository.userExists
    }

    fun getUserToken (uid : String) : LiveData<String?> {
        return loginRepository.getJwtToken(uid)
    }

    val getAllBanners : LiveData<HomePageBannerResponseModel>? = repository?.homePageBanners
}